odl/solvers/util/callback.py

```diff
@@ -1034,6 +1045,14 @@ def __repr__(self):
             return '{}({})'.format(self.__class__.__name__,
                                    inner_str)
 
+def _setupShouldEvaluateAtStep(self, step):
+    try:
+        self.step = frozenset(int(i) for i in step)
+        self.should_evaluate_at_step = lambda i: i in self.step
```

Comment:
I wanted to refactor the whole repetition and thought about returning self, so that the user does see that self is modified inside of this function. Otherwise there is a lot of code repetition and the inits are quite bloated if the user wants to take a look at them.
I also thought about a metaclass that will automatically setup the the init and call, but I guess that is a bit of an overkill